nixos/modules/services/web-servers/vencloud.nix

```
@@ -0,0 +1,100 @@
+{
+  lib,
+  pkgs,
+  config,
+  ...
+}:
+
+with lib;
+
+let
+  cfg = config.services.vencloud;
+in
+{
+  meta.doc = ./vencloud.md;
+  meta.maintainers = with lib.maintainers; [ eveeifyeve ];
+
+  options.services.vencloud = {
+    enable = mkEnableOption "Selfhosted vencloud api";
+
+    package = mkPackageOption pkgs "vencloud" { };
+
+    settings = mkOption {
+      type = types.submodule {
+        freeformType =
+          with types;
+          attrsOf (
+            nullOr (oneOf [
+              bool
+              int
+              port
+              str
+            ])
+          );
+        options = {
+          DISCORD_CLIENT_ID = mkOption {
+            type = types.nullOr types.int;
+            default = null;
+            example = 321436435432;
+            description = "The client id that should be used for discord OAuth";
+          };
+
+          DISCORD_CLIENT_SECRET = mkOption {
+            type = types.nullOr types.str;
+            default = null;
+            example = 352364526452;
+            description = "The client id that should be used for discord OAuth";
+          };
+
+          DISCORD_REDIRECT_URI = mkOption {
+            type = types.nullOr types.str;
+            default = "http://localhost:8000/v1/oauth/callback";
+            example = "https://example.com/v1/oauth/callback";
+            description = "The redirect url that should be used for discord OAuth";
+          };
+
+          HOST = mkOption {
+            type = types.nullOr types.str;
+            default = "0.0.0.0";
+            description = "The host that vencloud should host at";
+          };
+
+          PORT = mkOption {
+            type = types.nullOr types.port;
+            default = null;
+            description = "The port that vencloud should host at";
+          };
+
+          REDIS_URI = mkOption {
+            type = types.nullOr types.str;
+            default = null;
+            description = "The redis db url that you want to use for vencloud";
+          };
+        };
+      };
+      default = { };
+
+      description = ''
+        Configuration for vencloud, see
+        <link xlink:href="https://github.com/Vencord/Vencloud/blob/main/.env.example"/>
+        for supported values.
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.services.vencloud = {
+      description = "A CloudSync API for Vencord";
```

```suggestion
      description = "Vencord's API for cloud settings sync";
```
From the upstream README